packages/core/test/types.test.ts

```diff
@@ -4357,3 +4358,49 @@ describe('createActor', () => {
     createActor(logic);
   });
 });
+
+describe('snapshot methods', () => {
+  it('should type infer actor union snapshot methods', () => {
+    const typeOne = setup({
+      types: {} as {
+        events: { type: 'one' };
+        tags: 'one';
+      }
+    }).createMachine({
+      initial: 'one',
+      states: {
+        one: {}
+      }
+    });
+    type TypeOneRef = ActorRefFrom<typeof typeOne>;
+
+    const typeTwo = setup({
+      types: {} as {
+        events: { type: 'one' } | { type: 'two' };
+        tags: 'one' | 'two';
+      }
+    }).createMachine({
+      initial: 'one',
+      states: {
+        one: {},
+        two: {}
+      }
+    });
+    type TypeTwoRef = ActorRefFrom<typeof typeTwo>;
+
+    const ref = createActor(typeTwo) as TypeOneRef | TypeTwoRef;
+    const snapshot = ref.getSnapshot();
+
+    snapshot.can({ type: 'one' });
+    // @ts-expect-error
+    snapshot.can({ type: 'three' });
```

Comment:
shouldn't we test `'one'` and `'two'` in all of those assertions? `'three'` is incompatible with either - it's still a valuable test since it might test that the argument isn't `any`. However, it would also be good to test that only a subset of the "inputs" is allowed here (`'one'` should be OK, and `'two'` should not be)